File: solutions/fortran/eliuds-eggs/1/eliuds_eggs.f90
module eliuds_eggs
  implicit none
contains

  integer function eggCount(number)
    integer :: number
    integer :: temp_number
    
    eggCount = 0
    temp_number = number
    
   
    do while (temp_number > 0)

      eggCount = eggCount + iand(temp_number, 1)
      
   
      temp_number = ishft(temp_number, -1)
    end do
    
  end function eggCount

end module eliuds_eggs

File: solutions/typescript/eliuds-eggs/1/eliuds-eggs.ts
export const eggCount = (displayValue: number): number => {
  let count = 0;
  let num = displayValue;

  while (num > 0) {
    // Проверяем, равен ли самый правый бит единице
    count += num & 1;
    // Сдвигаем числа вправо на один бит
    num = num >>> 1;
  }

  return count;
};
